cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -232,6 +236,13 @@ public void stop() {
   }
 
   /**
+   * Resolve the scope of the user arguments from the {@link ProgramOptions}
+   */
+  protected Arguments resolveScope(ProgramOptions programOptions) {
```

Comment:
I think it's better for this method to take an `Arguments` and returns an `Arguments`.